src/libraries/System.Threading.Channels/src/System/Threading/Channels/SingleConsumerUnboundedChannel.cs

```diff
@@ -117,7 +117,7 @@ public override ValueTask<T> ReadAsync(CancellationToken cancellationToken)
                     parent._blockedReader = newBlockedReader;
                 }
 
-                oldBlockedReader?.TrySetCanceled();
+                oldBlockedReader?.TrySetCanceled(cancellationToken);
```

Comment:
This should use `default`.  It's being canceled not because of the cancellation token, but because the caller made a mistake and made another call on a "single reader" channel while another read was in progress, so we're canceling the previous read.